datasets.py

```diff
@@ -117,6 +122,84 @@ def read_xml_content(xml_file):
+    nlp_supercat = []
+    for supercat in supercat_list:
+        nlp_supercat.append(nlp(supercategories_to_names.get(supercat)))
+    ######################################################################  
+    # function that calculates the "closest" supercategory to input word
+    # and returns said supercategory and the associated softmax confidence score
+    def dist_calculator(word):
+        assert(word is not None)
+        # score array holding distance btwn word and each supercat, for softmax calculation
+        score_arr = []
+        for supercat_token in nlp_supercat:
+            cur_score = supercat_token.similarity(nlp(word))
+            score_arr.append(cur_score)
+        score_arr = softmax(score_arr)
+        return supercat_list[np.argmax(score_arr)], np.max(score_arr)
+    ######################################################################    
+    # this represents the result map from label to supercat
+    result_label_to_group_map = OrderedDict()
+    # (hr_label) is the human-readable value corresponding to the key (label)
+    for label, hr_label in labels_to_names.items():
+        if override_map and hr_label in override_map:
+            # skip if user is overriding with override_map
+            continue
+        # retrieve tuple result from dist_calculator
+        supercat_match = dist_calculator(hr_label)
+        result_label_to_group_map[label] = supercat_match
+        
+    # sort from least to most confident based on softmax scores
+    result_label_to_group_map = OrderedDict(sorted(result_label_to_group_map.items(), key=lambda item: item[1][1]))
+    
+    # remove the softmax scores from the dictionary
+    for k, v in result_label_to_group_map.items():
+        result_label_to_group_map[k] = v[0]
+    
+    # add the harded-coded override_map values if provided:
+    if override_map:
+        # initialize non-human-readable form of override_map
+        override_nonhuman_readable = {}
+        # invert the labels_to_names 
+        names_to_labels = dict((v, k) for k, v in labels_to_names.items())
+        # invert the supercategories_to_names
+        names_to_supercategories = dict((v, k) for k, v in supercategories_to_names.items())
+        # convert human-readable labels and supercats 
+        for k,v in override_map.items():
+            label = names_to_labels.get(k)
+            supercat = names_to_supercategories.get(v)
+            assert label in labels_to_names and supercat in supercategories_to_names, "override string not valid"
+            override_nonhuman_readable[label] = supercat
+        # add overriden dict to result dict 
+        result_label_to_group_map.update(override_nonhuman_readable)
+
+    # print mapping in human-readable form so user can adjust if necessary
+    print("Here are the 20 LEAST confident labels to supercategory mappings, ranked in increasing confidence.\nChange as necessary using override_map)")
```

Comment:
maybe change phrasing to be something like "here are 20 of the least confident...." so it doesn't seem like this specific 20 was picked for a reason, when this number was actually arbitrary